### PR TITLE
Adapting to new config dataProcessing

### PR DESCRIPTION
**Pull request by JuanluOnieva-biomage**
**Link to issue**

https://biomage.atlassian.net/browse/BIOMAGE-660?atlOrigin=eyJpIjoiZjYxOTY1MTJmZWU2NDZlOWEwMDAxY2Y5N2NjZmQ4MmQiLCJwIjoiaiJ9

**Anything else the reviewers should know about the changes here**

There are several points to deal with:

- Bioinformatics: I commented the filters that are now working in production (Step 1 to 5)
- Bioinformatics: I updated the * CreateSeuratObject* function to use min.cells=1 and min.features=1 because in 1-preproc.r we filtered by this parameter and it does not necessary to re-do it again.
- Bioinformatics: Some QC plots have been added by Sebastian. 
- Bioinformatics: Classifier is disabled in the case that the input dataset is filtered.
- Engineer: I have run some experiment to test it. An example of the processConfig is in the dynamo with the id e8bcf6a47618e58cc9b60175b40add72 (unisample experiment) and multisample_unfiltered (is the same as e52b39624588791a7889e39c617f669e)
- Engineer: I couldn't upload to dynamo the samples-table due to some permission that need to be reviewed. But for the review of the current PR [here](https://drive.google.com/file/d/1rOIuqIeIWaFncTLH2ghW7wSs0scqE_Ql/view?usp=sharing) and [here](https://drive.google.com/file/d/1CFoMx-Uhrmhsj_AvsV0B4SlNWGpqQZsG/view?usp=sharing) you can find examples of unisample and multisample experiments.

### CODE REVIEW

File: src/3-preproc.r [hunk 1]
```diff
@@ -103,8 +104,9 @@ config.mitochondrialContent <- list(enabled="true", auto="true",
     )
 )
 
-config.classifier <- list(enabled="true", auto="true", 
-    filterSettings = list(minProbability=0.82, bandwidth=-1, filterThreshold=-1)
+config.classifier <- list(enabled=tolower(as.character(!seurat_obj@tools$flag_filtered)) # emptyDrops results not present
+    , auto="true", 
+    filterSettings = list(FDR=0.1)
```
Comment (seb-mueller) [inline src/3-preproc.r:109]: suggestion to engineers: 
- add flag_filter to experiment config
- inform enable flag by this flag_filter ?

**Review by seb-mueller — APPROVED** [review]

**Comment by ivababukova** [comment]
why are you trying to adapt data-ingest to the new config when we will be very soon removing the data processing part from data ingest altogether and use the platform for data processing?

**Comment by ivababukova** [comment]
Or are these changes actually doing this: https://biomage.atlassian.net/browse/BIOMAGE-660

File: src/4-upload-to-aws.py [hunk 2]
```diff
@@ -20,6 +22,101 @@ def calculate_checksum(filenames):
...
+            "mime": mime_options[sample_file.split(".")[-1]],
+            "success": True, 
+            "error": False
+        }
+    samples_table["sample1"] = {
+        "name" : "sample1", 
+        "uuid" : uuid, 
+        "species": config["organism"],
+        "type": config["input"]["type"],
+        "cteatedData": createdDate.isoformat(),
```
Comment (aerlaut) [inline src/4-upload-to-aws.py:109]: There's typo in the key, it should be `createdDate`

**Review by seb-mueller — APPROVED** [review]

File: src/4-upload-to-aws.py [hunk 3]
```diff
@@ -20,6 +22,101 @@ def calculate_checksum(filenames):
...
+def create_samples_table_multisample(config, experiment_id, uuid):
+    # In samples_table we are going to add the core of the information
+    samples_table = {}
+    # Firstly, we identify the samples name. To do that we fetch the names of the folders (we suppose that the name
+    # of the folders corresponds with the samples name)
+    samples = [ name for name in os.listdir("/input") if os.path.isdir(os.path.join("/input", name)) ]
+    samples_table["ids"] = [ sample for sample in samples]
+    # We crate a cnt variable to track the number of samples, since the key value of the is: sample1, sample2, ... samplen 
+    # (with n as the total number of samples)
+    cnt = 1
```
Comment (aerlaut) [inline src/4-upload-to-aws.py:36]: I don't think you need this variable anymore

File: src/4-upload-to-aws.py [hunk 4]
```diff
@@ -20,6 +22,101 @@ def calculate_checksum(filenames):
...
+            "type": config["input"]["type"],
+            "createdDate": createdDate.isoformat(),
+            "lastModified": lastModified.isoformat(),
+            "complete": True, 
+            "error": False,
+            "fileNames" : sample_files, 
+            "files" : fileNames
+        }
+        # Incremen the iterator for the sampleX
+        cnt = cnt+1
```
Comment (aerlaut) [inline src/4-upload-to-aws.py:74]: also delete this if you delete `cnt`

**Pull request merged**